core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/V4CanonicalRequest.java

```diff
@@ -181,7 +181,7 @@ public static List<Pair<String, List<String>>> getCanonicalHeaders(Map<String, L
      * Each header-value pair is separated by a newline.
      */
     public static String getCanonicalHeadersString(List<Pair<String, List<String>>> canonicalHeaders) {
-        StringBuilder result = new StringBuilder(512);
+        StringBuilder result = new StringBuilder(2048);
```

Comment:
Can we add a note about why we chose this number?

Reply:
Done